app/services/creators/employments_creator.rb

```diff
@@ -9,12 +9,20 @@ def initialize(assessment_id:, employments_attributes: [])
     end
 
     def call
-      create_records
+      if json_validator.valid?
+        create_records
+      else
+        self.errors = json_validator.errors
+      end
       self
     end
 
   private
 
+    def json_validator
+      @json_validator ||= JsonValidator.new("employment", employments_incomes)
```

Comment:
other PRs have been passing raw JSON params in here i think. This is passing a hash. surprised it is not causing a  causing a problem and perhaps we should be using a single pattern?

Reply:
this is just the code that made me investigate further and discuss

Reply:
use @employment_params here